notesapp/models.py

```diff
@@ -0,0 +1,154 @@
+from django.db import models
+from django.db import transaction
+from django.contrib.auth.models import User
+from courses.models import(KnowledgeBase,
+						   SoftSkillsData,
+						   RandomData)
+from courses.utils.modelsutils import RowInformation
+from courses.utils.modelsutils import pl_custom_slugify
+
```

Comment:
from django.db import IntegrityError
please check whether how your app responds correctly to slug already exists
for eg.
note 1
title = something new
slug = something-new

note 2
title = something-new
slug = something-new-random1234  (expected)

currently, I'm getting an error because you didn't import IntegrityError